src/main/types.ts

```diff
@@ -157,6 +159,29 @@ interface ISettingsPanelProps {
     onChange: (event: IOurChangeEvent) => void,
 };
 
+/**
+ * Metadata type. 
+ */
+ type Metadata = {
+    name: string;
+    category: VideoCategory;
+    duration: number;
+    result: boolean;
+    zoneID?: number;
+    encounterID?: number;
+    difficultyID?: number;
+    playerName?: string;
+    playerRealm?: string;
+    playerSpecID?: number;
+    teamMMR?: number;
+    challengeMode?: ChallengeModeDungeon;
+    playerDeaths?: PlayerDeathType[];
+    chests?: number;
+    mapID?: number;
+    timeline?: ChallengeModeTimelineSegment[];
```

Comment:
This is already present in `ChallengeModeDungeon` - if you want to decouple it from that, you should refactor it out entirely and rename it, probably.

Reply:
my plan was to pull it out as including the entire ChallengeModeDungeon object in the metadata seems a bit odd I thought, open to being convinced either way

Reply:
(especially as I'm changing it as part of this PR - and might change it in future too)

Reply:
I think it's totally fine to pull it out of CM and let it be generic as I _did_ want to maybe use it for raids too to have boss phases specified, although after looking through the source code for WCL that is a really time consuming task as it's usually a case of boss HP, boss cast, or some other weird thing that separates phases.